src/formatting.rs

```diff
@@ -56,7 +56,12 @@ impl<'a> Highlighter<'a> {
     /// Write a single piece of highlighted text.
     pub fn highlight(&mut self, text: &str) -> io::Result<()> {
         for line in LinesWithEndings::from(text) {
-            for (style, component) in self.highlighter.highlight(line, self.syntax_set) {
+            // The following unwrap can only be removed with a breaking change to this functions return type.
+            for (style, component) in self
+                .highlighter
+                .highlight_line(line, self.syntax_set)
+                .unwrap()
+            {
```

Comment:
```suggestion
            for (style, component) in self
                .highlighter
                .highlight_line(line, self.syntax_set)
                .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?
            {
```
In the long term we should probably switch to anyhow for these but that has to be propagated in a lot of places.

Reply:
Umm, it isnt an `io::Error`.

Anyway, feel free to add your suggestion if you feel it is appropriate.

Reply:
Questionable things get folded into `io::Error` even in the stdlib. Usually as [`ErrorKind::InvalidData`](https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.InvalidData).

It's not ideal but I'd rather have a controlled crash than a panic.